### PR TITLE
add golines pre-commit hook

### PR DESCRIPTION
**Pull request by hmajid2301**
I tried to copy the gofmt and move it over to golines as the tooling is very similar. 
https://github.com/segmentio/golines

### CODE REVIEW

File: modules/hooks.nix [hunk 1]
```diff
@@ -2487,6 +2501,31 @@ lib.escapeShellArgs (lib.concatMap (ext: [ "--ghc-opt" "-X${ext}" ]) hooks.ormol
         # all file names in a single run.
         require_serial = true;
       };
+      golines =
+        {
+          name = "golines";
+          description = "A golang formatter that fixes long lines";
+          package = tools.golines;
```
Comment (sandydoo) [inline modules/hooks.nix:2508]: You have to import `golines` from pkgs and add it to tools. See `tools.nix`.

Comment (sandydoo) [inline modules/hooks.nix:2508]: You have to import `golines` from pkgs and add it to tools. See `tools.nix`.

**Review by sandydoo — APPROVED** [review]

**Review by sandydoo — CHANGES_REQUESTED** [review]
You have to import `golines` from pkgs and add it to tools. See `tools.nix`.

**Comment by hmajid2301** [comment]
I have made the change, please let me know if I did it correctly? Thanks

File: modules/hooks.nix [hunk 2]
```diff
@@ -565,6 +565,20 @@ in
           };
         };
       };
+      golines = mkOption {
+        description = "golines hook";
+        type = types.submodule {
+          imports = [ hookModule ];
+          options.settings = {
+            flags = mkOption {
```
Comment (sandydoo) [inline modules/hooks.nix:573]: For what it's worth, all hooks accept an `args` option that does this.

Comment (hmajid2301) [reply modules/hooks.nix:573]: ahh good to know for the next one thanks

**Pull request merged**